### PR TITLE
added Pagination, Sorting and Swgger

### PR DESCRIPTION
**Pull request by irynamekh**

### CODE REVIEW

File: src/main/java/com/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -15,34 +18,40 @@
...
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Get book by id")
```
Comment (padu12) [inline src/main/java/com/bookstore/controller/BookController.java]: I think we don't need this duplication of text in summary and description. Its better to remove one of them. Please do it for every next method

File: src/main/java/com/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -11,12 +12,15 @@
 @Data
 public class CreateBookRequestDto {
     @NotNull
+    @NotBlank
```
Comment (padu12) [inline src/main/java/com/bookstore/dto/CreateBookRequestDto.java:15]: You don't need both `@NotNull` and `@NotBlank` annotations. Please remove one of them and do it for all fields

**Review by padu12 — APPROVED** [review]
Amazing implementation of pagination and swagger! Lets improve some small things)

File: pom.xml [hunk 3]
```diff
@@ -57,8 +57,14 @@
             <version>${mapstruct.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.hibernate.validator</groupId>
-            <artifactId>hibernate-validator</artifactId>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+            <version>3.1.0</version>
```
Comment (nacenik) [inline pom.xml]: do you really need a version here?

File: pom.xml [hunk 4]
```diff
@@ -57,8 +57,14 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+            <version>2.1.0</version>
```
Comment (nacenik) [inline pom.xml]: I'd recommend you to put this to the `properties`

File: src/main/java/com/bookstore/controller/BookController.java (hunk 1 above)
Comment (nacenik) [reply src/main/java/com/bookstore/controller/BookController.java]: Good point, you can put to the summary just sth like this: 
- Get a book
- Get books
- Create a book
- Update a book
- etc.

File: src/main/java/com/bookstore/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (nacenik) [reply src/main/java/com/bookstore/dto/CreateBookRequestDto.java:15]: not agree. NotNull and NotBlank validate different things. Better use both of these annotations.

File: src/main/java/com/bookstore/repository/BookRepository.java [hunk 5]
```diff
@@ -1,7 +1,11 @@
 package com.bookstore.repository;
 
 import com.bookstore.model.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
+
+    Page<Book> findAll(Pageable pageable);
```
Comment (nacenik) [inline src/main/java/com/bookstore/repository/BookRepository.java]: I think JpaRepository has already had this method, and you don't need this bu your hand to your repository. Please double check this

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it a little bit :)

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**